### DISCUSSION TITLE
Custom Presidio Deployment - Azure App Service (Linux)

### DISCUSSION THREAD

**Discussion by SanthoshPK** [microsoft/presidio#869] [Q&A]
Hi All - 

We have used Presidio analyser and ananymizer modules in one simple Python APP (With only 2 API methods - 1 for analyse and 2 for anonymize). 

Installing dependencies with requirement.txt as below - 
flask
presidio_analyzer
presidio_anonymizer
presidio_image_redactor
https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.2.0/en_core_web_lg-3.2.0.tar.gz#egg=en_core_web_lg==3.2.0 (This works but it disappears after some time)
Or used 
python -m spacy download en_core_web_lg (This does not work )
also, We installed spacy model directly in Portal via SSH but still it installs and disappears .

In Azure - We have used B1 SKU with Python 3.9 stack for api deployment which becomes not accessible when related dependencies goes missing .

Please advise and any help will be appreciated :)

**Comment by omri374** [comment]
Hi @SanthoshPK, could you please provide more detail on what didn't work with the spaCy model installation? Is this where the issue is? Could it be an issue with spaCy itself? This is how we install the models within a Docker container: https://github.com/microsoft/presidio/blob/main/presidio-analyzer/install_nlp_models.py

Perhaps you could re-use some of this in your installation (`python install_nlp_models.py --conf_file conf/default.yaml`)

**Reply by SanthoshPK** [reply]
Thanks for the reply @omri374 . 

We have the requirements file with below value - 
https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.2.0/en_core_web_lg-3.2.0.tar.gz#egg=en_core_web_lg==3.2.0

Once we deploy the Python API code to Azure App Service via VS Code - It does complete the installation of spacy model .
And it can be seen in App Service - Like using SSH into Linux host and seen in pip freeze .

But it disappears after some time (random)  after disappears again i tried installing like below 

(antenv) root@2a379a211b4c:/tmp/8da3e2a15e2a266# pip install https://github.com/explosion/spacy-models/releases/download/en_core_web_lg-3.2.0/en_core_web_lg-3.2.0.tar.gz#egg=en_core_web_lg==3.2.0

Still same issue happens .
So not sure its issue with deployment of spacy models or issues with azure app service itself . 
With the model API does not work has Presidio analyser internally uses the spacy models.

also, we are not using the docker just plain simple API which uses modules of Presidio in it and it will be deployed to app service in azure as api flask app .

thanks .

**Reply by balteravishay** [reply]
@SanthoshPK could this be related to the App Service tier you are using (App Service B1 plan), which has only 1.75GB RAM? perhaps try to scale the plan up and validate if this is working for you?
